src/djcytoscape/views.py

```diff
@@ -137,54 +132,59 @@ def primary(request):
-    if request.method == 'POST':
-        form = GenerateQuestMapForm(request.POST)
-        # check whether it's valid:
-        if form.is_valid():
-            ct = form.cleaned_data['initial_content_type']
-            obj_id = form.cleaned_data['initial_object_id']
-            scape = form.cleaned_data['scape']
-            name = form.cleaned_data['name']
-            obj = ct.get_object_for_this_type(id=obj_id)
-            if not name:
-                name = str(obj)
-            new_scape = CytoScape.generate_map(initial_object=obj, name=name, parent_scape=scape, autobreak=autobreak)
-            return redirect('djcytoscape:quest_map', scape_id=new_scape.id)
-
-    else:
-
-        initial = {}
-        if ct_id and obj_id:
-            ct = get_object_or_404(ContentType, pk=ct_id)
-            initial['initial_content_type'] = ct
-            initial['initial_object_id'] = obj_id
-        if scape_id:
-            scape = get_object_or_404(CytoScape, pk=scape_id)
-            initial['scape'] = scape
-            interlink = True
-
-        form = GenerateQuestMapForm(initial=initial)
-
-    context = {
-        'form': form,
-        'interlink': interlink,
-    }
-
-    return render(request, 'djcytoscape/generate_new_form.html', context)
+class ScapeGenerateMap(NonPublicOnlyViewMixin, FormView):
+    form_class = GenerateQuestMapForm
+    template_name = 'djcytoscape/generate_new_form.html'
+    
+    @method_decorator(staff_member_required)
+    def dispatch(self, *args, **kwargs):
+        """
+        :param request:
+        :param ct_id: Content Type for Generic Foreign Key (initial object)
+        :param obj_id: Object ID for Generic Foreign Key (initial object)
+        :param scape_id: originating scape/quest
+        :return:
+        """
+        self.ct_id = kwargs.get('ct_id')
+        self.obj_id = kwargs.get('obj_id')
+        self.scape_id = kwargs.get('scape_id')  
+        self.autobreak = kwargs.get('autobreak', True)
+        self.interlink = False
+
+        return super().dispatch(*args, **kwargs)
+    
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs().copy()
+        kwargs['autobreak'] = self.autobreak
+        
+        if self.ct_id and self.obj_id:
+            ct = get_object_or_404(ContentType, pk=self.ct_id)
+            kwargs['initial'].update({
+                'initial_content_type': ct,
+                'initial_object_id': self.obj_id,
+            })
+
+        if self.scape_id:
+            scape = get_object_or_404(CytoScape, pk=self.scape_id)
+            kwargs['initial']['scape'] = scape
+            self.interlink = True
+
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs).copy()
+        ctx['interlink'] = self.interlink
+        return ctx
+
+    def form_valid(self, form):  # virtually the same as super().form_valid except we're caching form.save() obj for get_success_url()
```

Comment:
Good comment

Reply:
Can you add a message in here? Something like:
```
        messages.success(
            self.request,
            f"New map {self.object.name} was successfully generated."
        )
```

Reply:
done